src/hacbs/components/ApplicationDetails/tabs/overview/visualization/hooks/useAppWorkflowData.ts

```diff
@@ -1,129 +1,582 @@
-        : ['component-integration-test'],
-    },
-    staticEnv: {
-      id: 'static-env',
-      data: {
-        label: 'environment',
-        workflowType: WorkflowNodeType.ENVIRONMENT,
-        isDisabled: environments.length === 0,
-        resources: environments,
-      },
-      runBefore: [],
-      runAfterResourceKey: 'spec.parentEnvironment',
-      runAfter: isResourcesAvailable(applicationIntegrationTests)
-        ? getNodeNames(applicationIntegrationTests)
-        : ['application-integration-test'],
-    },
-    releasePlan: {
-      id: 'release-links',
-      data: {
-        label: 'managed environment',
-        workflowType: WorkflowNodeType.ENVIRONMENT,
-        isDisabled: releasePlans.length === 0,
-        resources: releasePlans,
-      },
-      runBefore: [],
-      runAfter: isResourcesAvailable(environments)
-        ? getLastEnvironments(environments)
-        : ['static-env'],
-    },
-  };
+  const applicationTestNodes = React.useMemo(() => {
+    if (!allResourcesLoaded) {
+      return [];
+    }
+    const nodes = applicationIntegrationTests?.length
+      ? applicationIntegrationTests.map((test) =>
+          resourceToPipelineNode(
+            test,
+            WorkflowNodeType.APPLICATION_TEST,
+            componentIntegrationTestTasks,
+          ),
+        )
+      : [
+          emptyPipelineNode(
+            'no-application-tests',
+            'Application tests',
+            WorkflowNodeType.APPLICATION_TEST,
+            componentIntegrationTestTasks,
+          ),
+        ];
+    const maxWidth = nodes.reduce((max, node) => Math.max(max, node.width), 0);
+    nodes.forEach((n) => (n.width = maxWidth));
+    return nodes;
+  }, [allResourcesLoaded, applicationIntegrationTests, componentIntegrationTestTasks]);
+
+  const applicationIntegrationTestTasks = React.useMemo(
+    () => (expanded ? applicationTestNodes.map((c) => c.id) : []),
+    [applicationTestNodes, expanded],
+  );
+
+  const testsGroup = React.useMemo(
+    () =>
+      allResourcesLoaded
+        ? groupToPipelineNode(
+            'tests',
+            'Tests',
+            WorkflowNodeType.TESTS,
+            buildTasks,
+            expanded,
+            expanded
+              ? [...componentIntegrationTestTasks, ...applicationIntegrationTestTasks]
+              : undefined,
+            [...componentTestNodes, ...applicationTestNodes],
+            [...componentIntegrationTests, ...applicationIntegrationTests],
+          )
+        : undefined,
+    [
+      allResourcesLoaded,
+      applicationIntegrationTestTasks,
+      applicationIntegrationTests,
+      applicationTestNodes,
+      buildTasks,
+      componentIntegrationTestTasks,
+      componentIntegrationTests,
+      componentTestNodes,
+      expanded,
+    ],
+  );
 
-  const nodes: WorkflowNode[] = workflowToNodes(workflowObject);
+  const staticEnvironments = environments.filter(
+    (e) => getEnvironmentType(e) === EnvironmentType.static,
+  );
```

Comment:
You need to use `useMemo` here since `staticEnvironments ` is used as a memo dependency. Otherwise the dependency check will always fail.